src/openms/source/CHEMISTRY/ISOTOPEDISTRIBUTION/CoarseIsotopePatternGenerator.cpp

```diff
@@ -90,6 +90,15 @@ namespace OpenMS
     for (; it != formula.end(); ++it)
     {
       IsotopeDistribution tmp = it->first->getIsotopeDistribution();
+
+      // filter out unstable isotopes / those with zero natural abundance as
+      // they just blow up the computations we have do to but have zero
+      // abundance
+      auto data = tmp.getContainer();
```

Comment:
do we need to copy here?
would auto&
work and omiting the tmp.set?

Reply:
I think we would have to add a non-const function for this

Reply:
Btw: this is not the slow part of the function :-)